modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/CurrencySnapshotConsensusFunctions.scala

```diff
@@ -61,7 +71,12 @@ object CurrencySnapshotConsensusFunctions {
       events: Set[CurrencySnapshotEvent]
     ): F[(CurrencySnapshotArtifact, CurrencySnapshotContext, Set[CurrencySnapshotEvent])] = {
 
-      val blocksForAcceptance = events
+      val (blocks: List[Signed[CurrencyBlock]], dataBlocks: List[Signed[DataApplicationBlock]]) = events.collect {
+        case event @ Left(_)                                           => event
+        case event @ Right(_) if maybeDataApplicationService.isDefined => event
+      }.toList.partitionMap(identity)
```

Comment:
`Left` case is just `a => a`. Also `Right` case is just a filter on `Right` side.

```scala
events
  .filter(_ => maybeDataApplicationService.isDefined)
  .toList
  .partitionMap(identity)
```

Reply:
I don't think filter is right biased here. I think it will filter out all events (lefts and rights) based on whether  dataApplicationService is none or some.

Reply:
@kpudlik: @TheMMaciek is right. `filter` on Either is right biased, but `events` are `Set[Either`. anyway, I've updated the code to use `filter` instead of `collect` which seems to be more readable

Reply:
`events.filter(_.isLeft || maybeDataApplicationService.isDefined)` then :D